lib/libshare/smb.c

```diff
@@ -138,10 +138,14 @@ smb_retrieve_shares(void)
 
 			if (strcmp(key, "path") == 0)
 				path = dup_value;
-			if (strcmp(key, "comment") == 0)
+			else if (strcmp(key, "comment") == 0)
 				comment = dup_value;
-			if (strcmp(key, "guest_ok") == 0)
+			else if (strcmp(key, "guest_ok") == 0)
 				guest_ok = dup_value;
+			else {
+				free(dup_value);
+				dup_value = NULL;
+			}
 
 			if (path == NULL || comment == NULL || guest_ok == NULL)
 				continue; /* Incomplete share definition */
```

Comment:
**share_file_fp, there may be two "key" line, there will be a memory leak.** 
For example:
In share_file_fp pointing to the content of the files
  ```
  comment=value1
  comment=value2
  path=value3
  path=value4
  guest_ok=2
  guest_ok=3
```
so, read file, then path , comment , guest_ok will be a memory leak.